Source/AtomicJS/Javascript/JSScene.cpp

```diff
@@ -24,6 +24,27 @@ static int Node_CreateJSComponent(duk_context* ctx)
     return 1;
 }
 
+static int Node_GetJSComponent(duk_context* ctx)
+{
+	StringHash classNameHash = duk_to_string(ctx, 0);
+	duk_push_this(ctx);
+	Node* node = js_to_class_instance<Node>(ctx, -1, 0);
+
+	PODVector<JSComponent*> jsComponents;
+	node->GetComponents<JSComponent>(jsComponents);
+
+	for (unsigned i = 0; i < jsComponents.Size(); i++)
+	{
+		if (StringHash(jsComponents[i]->GetClassName()) == classNameHash)
+		{
+			js_push_class_object_instance(ctx, jsComponents[i]);
+			break;
+		}
+	}
+
+	return 1;
+}
+
```

Comment:
Ok, so a couple things here:

1) There may be more than one JS component on a node, this only gets the 1st

2)  This function should be unnecessary as the node object is populated with a property of the component class:

https://github.com/AtomicGameEngine/AtomicGameEngine/blob/master/Source/AtomicJS/Javascript/JSVM.cpp#L177

.. and there is a lookup to all components of that class on  the node in the components lookup on the node:

https://github.com/AtomicGameEngine/AtomicGameEngine/blob/master/Source/AtomicJS/Javascript/JSVM.cpp#L180

This should all be documented, sorry.  

I also think that the JSComponents will be able to be improved with some of the new scripting features, like being able in use inheritance on a prototype chain from script.